### PR TITLE
Add filter panel sections

### PR DESCRIPTION
**Pull request by barrymcgee**
- Run Storybook
- http://localhost:65233/?path=/story/searchandfilter--external-state

Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1327,
Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1329

### CODE REVIEW

**Comment by anthonydillon** [comment]
After a quick chat this would be better as a sub-component.

File: src/components/SearchAndFilter/FilterPanelSection/FilterPanelSection.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import "./filter-panel-section.scss";
+
+const FilterPanelSection = ({ data }) => (
+  <div className="filter-panel-section">
+    {data?.heading && (
+      <h3 className="filter-panel-section__heading">{data.heading}</h3>
+    )}
+    {data?.chips?.map((chip, i) => (
+      <Chip lead={chip.lead} value={chip.value} key={i} />
```
Comment (hatched) [inline src/components/SearchAndFilter/FilterPanelSection/FilterPanelSection.js:13]: If the order of the chips in the `data` changes then `key` will become invalid and trigger a re-render. It'll be a bit more performant if the key is tied to the representation of the chip, so maybe...

```suggestion
      <Chip lead={chip.lead} value={chip.value} key={chip.lead+chip.value} />
```

File: src/components/SearchAndFilter/FilterPanelSection/FilterPanelSection.test.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { shallow, mount } from "enzyme";
+import React from "react";
+
+import FilterPanelSection from "./FilterPanelSection";
+
+const sampleData = {
+  heading: "Regions",
+  chips: [{ value: "us-east1" }, { value: "us-east2" }, { value: "us-east3" }],
```
Comment (hatched) [inline src/components/SearchAndFilter/FilterPanelSection/FilterPanelSection.test.js:8]: Sample data doesn't contain `lead` nor do any of the tests.

File: src/components/SearchAndFilter/SearchAndFilter.test.js [hunk 3]
```diff
@@ -3,6 +3,25 @@ import React from "react";
 
 import SearchAndFilter from "./SearchAndFilter";
 
+const sampleData = [
+  {
+    heading: "Clouds",
+    chips: [{ value: "Google" }, { value: "AWS" }, { value: "Azure" }],
```
Comment (hatched) [inline src/components/SearchAndFilter/SearchAndFilter.test.js:9]: missing `lead` in this data set as well.

File: src/components/SearchAndFilter/FilterPanelSection/FilterPanelSection.test.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  });
+
+  it("shows correct data passed as prop", () => {
+    const wrapper = mount(<FilterPanelSection data={sampleData} />);
+    expect(wrapper.find(".filter-panel-section").length).toEqual(1);
+    const section = wrapper.find(".filter-panel-section");
+    expect(section.find(".filter-panel-section__heading").text()).toEqual(
+      "Regions"
+    );
+    const firstChip = section.find(".p-chip").at(0);
```
Comment (anthonydillon) [inline src/components/SearchAndFilter/FilterPanelSection/FilterPanelSection.test.js:24]: Can you find it via the key? Therefore the order will not matter?

File: src/components/SearchAndFilter/FilterPanelSection/FilterPanelSection.test.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    expect(firstChip.text()).toEqual("us-east1");
```
Comment (anthonydillon) [inline src/components/SearchAndFilter/FilterPanelSection/FilterPanelSection.test.js:25]: I would expect a count check on the chip elements.

File: src/components/SearchAndFilter/SearchAndFilter.test.js (hunk 3 above)
Comment (anthonydillon) [reply src/components/SearchAndFilter/SearchAndFilter.test.js:9]: The Chip element itself has tests to cover lead and non-lead so testing that here is probably not needed.

**Pull request merged**